specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2022-07-01-preview/healthcare-apis.json

```diff
@@ -4093,6 +4138,224 @@
+        "properties": {
+          "type": "array",
+          "x-ms-identifiers": [],
+          "items": {
+            "$ref": "#/definitions/MedtechMeasurementProperty"
+          }
+        }
+      },
+      "readOnly": true
+    },
+    "MedtechMappingValidationLineInfo": {
+      "type": "object",
+      "description": "Associates the validation error with the position on the mapping template where it occurred ",
+      "readOnly": true,
+      "properties": {
+        "lineNumber": {
+          "description": "The line on which the validation error was discovered",
+          "type": "integer",
+          "format": "int32"
+        },
+        "linePosition": {
+          "description": "The position on the line which the validation error was discovered",
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "MedtechMappingValidationCategory": {
+      "type": "string",
+      "description": "Indicates the type of validation being performed",
+      "enum": [
+        "NORMALIZATION",
+        "FHIRTRANSFORMATION"
+      ],
+      "x-ms-enum": {
+        "name": "MedtechMappingValidationCategory",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "MedtechMappingValidationErrorLevel": {
+      "type": "string",
+      "description": "Indicates the severity of the validation error",
+      "enum": [
+        "ERROR",
+        "WARN"
+      ],
+      "x-ms-enum": {
+        "name": "MedtechMappingValidationErrorLevel",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "MedtechMappingValidationError": {
+      "type": "object",
+      "description": "A mapping validation error.",
+      "readOnly": true,
+      "properties": {
+        "message": {
+          "description": "The validation error message.",
+          "type": "string"
+        },
+        "level": {
+          "$ref": "#/definitions/MedtechMappingValidationErrorLevel"
+        },
+        "category": {
+          "$ref": "#/definitions/MedtechMappingValidationCategory"
+        },
+        "lineInfo": {
+          "$ref": "#/definitions/MedtechMappingValidationLineInfo"
+        }
+      }
+    },
+    "MedtechMappingValidationErrorList": {
+      "type": "array",
+      "description": "A list of mapping validation error.",
+      "x-ms-identifiers": [],
+      "readOnly": true,
+      "items": {
+        "$ref": "#/definitions/MedtechMappingValidationError"
+      }
+    },
+    "ValidateMedtechMappingsDeviceResult": {
+      "description": "A result returned when processing a single device event in Medtech Mapping Validation Service.",
+      "type": "object",
+      "properties": {
+        "aggregatedCount": {
+          "description": "Indicates how many Device Events produced the associated Exception.",
+          "type": "object"
+        },
+        "deviceEvent": {
+          "description": "The device event which was validated.",
+          "type": "object"
+        },
```

Comment:
free form objects are not allowed , please define these objects explicitly

Reply:
This data is supplied by the end user. Other than being a JSON object, there isn't a schema for it. The customer supplies us with JsonPath expressions which we will then use to extract out values from the data that they supply.  Given this, can we make an exception here and keep this as type `object`?

Reply:
Please check my other comment for freeform data